crates/modelardb_common/src/arguments.rs

```diff
@@ -121,3 +168,138 @@ pub async fn validate_remote_data_folder(
+            .build()
+            .map_err(|error| Status::invalid_argument(error.to_string()))?,
+    );
+
+    validate_remote_data_folder(RemoteDataFolderType::S3, &s3)
+        .await
+        .map_err(Status::invalid_argument)?;
+
+    Ok(s3)
+}
+
+/// Parse the arguments in `data` and return an [`Azure Blob Storage`](object_store::azure::MicrosoftAzure)
+/// object store if `data` contains the necessary arguments. If `data` is missing arguments or if the created
+/// [`Azure Blob Storage`](object_store::azure::MicrosoftAzure) object store connection is invalid,
+/// [`Status`] is returned.
+pub async fn parse_azure_blob_storage_arguments(
+    data: &[u8],
+) -> Result<Arc<dyn ObjectStore>, Status> {
+    let (account, offset_data) = extract_argument(data)?;
+    let (access_key, offset_data) = extract_argument(offset_data)?;
+    let (container_name, _offset_data) = extract_argument(offset_data)?;
+
+    let azure_blob_storage: Arc<dyn ObjectStore> = Arc::new(
+        MicrosoftAzureBuilder::new()
+            .with_account(account)
+            .with_access_key(access_key)
+            .with_container_name(container_name)
+            .build()
+            .map_err(|error| Status::invalid_argument(error.to_string()))?,
+    );
+
+    validate_remote_data_folder(RemoteDataFolderType::AzureBlobStorage, &azure_blob_storage)
+        .await
+        .map_err(Status::invalid_argument)?;
+
+    Ok(azure_blob_storage)
+}
+
+/// Convert the given `argument` into padded bytes that contain the length of the byte
+/// representation of `argument` together with the byte representation. The length is exactly two
+/// bytes long.
+pub fn encode_argument(argument: &str) -> Vec<u8> {
+    let argument_bytes: Vec<u8> = argument.as_bytes().into();
+    let mut argument_size_bytes = vec![0; 2];
+
+    // unwrap() is safe since the buffer is in memory and no I/O errors can occur.
+    argument_size_bytes
+        .write_all(&argument_bytes.len().to_be_bytes())
+        .unwrap();
+
+    [
+        &argument_size_bytes[(argument_size_bytes.len() - 2)..],
+        argument_bytes.as_slice(),
+    ]
+    .concat()
+}
+
+/// Assumes `data` is a slice containing one or more arguments with the following format:
+/// size of argument (2 bytes) followed by the argument (size bytes). Returns a tuple containing
+/// the first argument and `data` with the extracted argument's bytes removed.
+pub fn extract_argument(data: &[u8]) -> Result<(&str, &[u8]), Status> {
+    let size_bytes: [u8; 2] = data[..2]
+        .try_into()
+        .map_err(|_| Status::internal("Size of argument is not 2 bytes."))?;
+
+    let size = u16::from_be_bytes(size_bytes) as usize;
+
+    let argument = std::str::from_utf8(&data[2..(size + 2)])
+        .map_err(|error| Status::invalid_argument(error.to_string()))?;
+    let remaining_bytes = &data[(size + 2)..];
+
+    Ok((argument, remaining_bytes))
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
```

Comment:
Added tests for `argument_to_connection_info()` but it was not possible to add tests for `parse_object_store_arguments()` since the remote data folder is validated as part of the function which means a valid S3 and/or Azure Blob Storage connection is required.